tests/unit/task_processor/test_unit_task_processor_processor.py

```diff
@@ -303,6 +303,40 @@ def _dummy_recurring_task() -> None:
         assert task_run.error_details is None
 
 
+@pytest.mark.django_db(transaction=True)
+def test_run_recurring_tasks_loops_over_all_tasks(
+    db: None,
+    run_by_processor: None,
+) -> None:
+    # Given, Three recurring tasks
+    @register_recurring_task(run_every=timedelta(milliseconds=200))
+    def _dummy_recurring_task_1() -> None:
+        pass
+
+    @register_recurring_task(run_every=timedelta(milliseconds=200))
+    def _dummy_recurring_task_2() -> None:
+        pass
+
+    @register_recurring_task(run_every=timedelta(milliseconds=200))
+    def _dummy_recurring_task_3() -> None:
+        pass
+
+    initialise()
+    task_runs = []
+
+    # When, we call run_recurring_tasks in a loop few times
+    for _ in range(4):
+        task_runs.append(run_recurring_tasks())
```

Comment:
We're not actually using the `task_runs` variable here, right? There are no actual assertions on it. So, we should either modify this section of code to look like: 

```suggestion
    initialise()

    # When, we call run_recurring_tasks in a loop few times
    for _ in range(4):
        run_recurring_tasks()
```

OR, we should add some assertions on the `task_runs` variable (perhaps instead of hitting the database again?)

Reply:
Done